nvflare/fuel/f3/cellnet/new_cell.py

```diff
@@ -0,0 +1,180 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import threading
+import time
+import uuid
+from typing import Dict, List, Union
+
+from nvflare.fuel.f3.cellnet.cell import Cell
+from nvflare.fuel.f3.cellnet.defs import MessageHeaderKey, MessageType
+from nvflare.fuel.f3.cellnet.registry import Registry
+from nvflare.fuel.f3.message import Message
+from nvflare.fuel.f3.stream_cell import StreamCell
+from nvflare.fuel.f3.streaming.stream_const import StreamHeaderKey, StreamMessageKey
+from nvflare.fuel.f3.streaming.stream_types import StreamFuture
+from nvflare.private.defs import CellChannel
+
+
+class Adapter:
+    def __init__(self, cb, my_info, nice_cell):
+        self.cb = cb
+        self.my_info = my_info
+        self.nice_cell = nice_cell
+        self.logger = logging.getLogger(self.__class__.__name__)
+
+    def call(self, future):  # this will be called by StreamCell upon receiving the first byte of blob
+        headers = future.headers
+        stream_req_id = headers.get(StreamMessageKey.REQ_ID, "")
+        origin = headers.get(MessageHeaderKey.ORIGIN, None)
+        print("received")
+        starting_time = time.time()
+        while not future.done():
+            self.logger.debug(
+                f"{future.get_progress()/1024/1024/1024:.3f} GiB, elapsed time: {(time.time()-starting_time):.1f} seconds"
+            )
+            time.sleep(10)
+        result = future.result()
+        request = Message(headers, result)
+        channel = request.get_header(StreamHeaderKey.CHANNEL)
+        request.set_header(MessageHeaderKey.CHANNEL, channel)
+        topic = request.get_header(StreamHeaderKey.TOPIC)
+        request.set_header(MessageHeaderKey.TOPIC, topic)
+        req_id = request.get_header(MessageHeaderKey.REQ_ID)
+        response = self.cb(request)
+        response.add_headers(
+            {
+                MessageHeaderKey.CHANNEL: channel,
+                MessageHeaderKey.TOPIC: topic,
+                MessageHeaderKey.FROM_CELL: self.my_info.fqcn,
+                MessageHeaderKey.ORIGIN: self.my_info.fqcn,
+                MessageHeaderKey.DESTINATION: origin,
+                MessageHeaderKey.TO_CELL: origin,
+                MessageHeaderKey.REQ_ID: req_id,
+                MessageHeaderKey.MSG_TYPE: MessageType.REPLY,
+                MessageHeaderKey.ROUTE: [(self.my_info.fqcn, time.time())],
+                StreamMessageKey.REQ_ID: stream_req_id,
+            }
+        )
+        messagesend_future = self.nice_cell.send_blob(channel, topic, origin, response)
+
+
+class NewCell(StreamCell):
+    def __init__(self, *args, **kwargs):
+        self.core_cell = Cell(*args, **kwargs)
+        super().__init__(self.core_cell)
+        self.req_reg = Registry()
```

Comment:
Should be able to use the registry in StreamCell

Reply:
This self.req_reg can be removed as it's not been used in this class.